source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/ActionLayoutRenderer.java

```diff
@@ -0,0 +1,126 @@
+package io.adaptivecards.renderer;
+
+import android.content.Context;
+import android.support.v4.app.FragmentManager;
+import android.view.Gravity;
+import android.view.ViewGroup;
+import android.widget.HorizontalScrollView;
+import android.widget.LinearLayout;
+
+import io.adaptivecards.objectmodel.ActionAlignment;
+import io.adaptivecards.objectmodel.ActionsOrientation;
+import io.adaptivecards.objectmodel.BaseActionElement;
+import io.adaptivecards.objectmodel.BaseActionElementVector;
+import io.adaptivecards.objectmodel.HostConfig;
+import io.adaptivecards.objectmodel.IconPlacement;
+import io.adaptivecards.objectmodel.Spacing;
+import io.adaptivecards.renderer.AdaptiveWarning;
+import io.adaptivecards.renderer.BaseCardElementRenderer;
+import io.adaptivecards.renderer.IActionLayoutRenderer;
+import io.adaptivecards.renderer.IBaseActionElementRenderer;
+import io.adaptivecards.renderer.RenderedAdaptiveCard;
+import io.adaptivecards.renderer.actionhandler.ICardActionHandler;
+import io.adaptivecards.renderer.registration.CardRendererRegistration;
+
+public class ActionLayoutRenderer implements IActionLayoutRenderer {
+
+    protected ActionLayoutRenderer()
+    {
+    }
+
+    public static ActionLayoutRenderer getInstance()
+    {
+        if (s_instance == null)
```

Comment:
new to this codebase -- is it safe to assume that this won't be called on two threads at once?